modules/fedimint-ln-server/src/lib.rs

```diff
@@ -1063,6 +1071,25 @@ impl Lightning {
         dbtx.get_value(&ContractKey(contract_id)).await
     }
 
+    async fn list_unspent_incoming_contracts(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+    ) -> Vec<ContractId> {
+        // Use `LightningAuditItemKey` to find incoming contracts with non-zero value,
+        // since these are cleaned up when the contract is claimed.
+        dbtx.find_by_prefix(&LightningAuditItemKeyPrefix)
+            .await
+            .filter_map(|(contract, _)| async move {
+                if let LightningAuditItemKey::Incoming(incoming) = contract {
+                    Some(incoming)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>()
+            .await
+    }
+
```

Comment:
Using the audit API for this makes the audit API harder to change later on, I'd like to avoid that.